packages/model/src/user.ts

```diff
@@ -218,15 +218,34 @@ export interface UserAvatarPostRequest {
 
 export type ListUserResponse = ListResponse<UserResponse>;
 
-export type FetchUsersFilter = {
-  role?: string[];
-  labId?: string[];
-  teamId?: string[];
-  code?: string;
-  hidden?: boolean;
-  onboarded?: boolean;
-  orcid?: string;
-};
+export type FetchUsersFilter =
+  | {
+      labId?: never;
+      role?: never;
+      teamId?: never;
+      code?: string;
+      hidden?: boolean;
+      onboarded?: boolean;
+      orcid?: string;
+    }
+  | {
+      labId?: string;
+      role?: never;
+      teamId?: never;
+      code?: never;
+      hidden?: never;
+      onboarded?: never;
+      orcid?: never;
+    }
+  | {
+      labId?: never;
+      role?: string[];
+      teamId?: string;
+      code?: never;
+      hidden?: never;
+      onboarded?: never;
+      orcid?: never;
+    };
 
 export type FetchUsersOptions = FetchOptions<FetchUsersFilter>;
```

Comment:
as mentioned previously I think you want to disable search entirely no? in which case I would advise to remove it from types too